### PR TITLE
Add more kubectl auth reconcile flags

### PR DESCRIPTION
**Pull request by mrogers950**
```release-note
The --remove-extra-subjects and --remove-extra-permissions flags have been enabled for kubectl auth reconcile
```
/assign @deads2k 
cc @enj @juanvallejo

### CODE REVIEW

File: pkg/kubectl/cmd/auth/reconcile.go [hunk 1]
```diff
@@ -89,6 +90,7 @@ func NewCmdReconcile(f cmdutil.Factory, streams genericclioptions.IOStreams) *co
 
 	cmdutil.AddFilenameOptionFlags(cmd, o.FilenameOptions, "identifying the resource to reconcile.")
 	cmd.Flags().BoolVar(&o.DryRun, "dry-run", o.DryRun, "If true, display results but do not submit changes")
+	cmd.Flags().BoolVar(&o.AdditiveOnly, "additive-only", true, "If true, preserve modified roles and bindings")
```
Comment (enj) [inline pkg/kubectl/cmd/auth/reconcile.go]: I assume the `true` should be set in `o` and then passed as the third arg?

**Comment by mrogers950** [comment]
Updated.

**Comment by enj** [comment]
> This needs to wait until 1.12

SGTM

> I prefer using CLI commands that bool flags default to false.

`--tighten`?

**Comment by mrogers950** [comment]
> --tighten?

how about `--restore` ?

**Comment by liggitt** [comment]
I think separate control over removing extra  permissions and removing extra subjects makes sense,  and makes for very clear flags 

--remove-extra-permissions
--remove-extra-subjects

**Comment by mrogers950** [comment]
Updated with separated flags as @liggitt suggested.

**Review by soltysh — APPROVED** [review]
/lgtm
/approve

**Pull request merged**